src/main/java/roomescape/repository/RoomescapeRepositoryImpl.java

```diff
@@ -0,0 +1,106 @@
+                SELECT r.id AS reservation_id, r.name, r.date, t.id AS time_id, t.start_at AS time_value
+                FROM reservation as r 
+                INNER JOIN reservation_time AS t
+                ON r.time_id = t.id
+                WHERE r.id = ?
+                """;
+        return template.queryForObject(sql, reservationRowMapper(), id);
+    }
+
+    @Override
+    public List<Reservation> findByDate(final LocalDate date) {
+        String sql = """
+                SELECT r.id AS reservation_id, r.name, r.date, t.id AS time_id, t.start_at AS time_value
+                FROM reservation as r 
+                INNER JOIN reservation_time AS t
+                ON r.time_id = t.id
+                WHERE r.date = ?
+                """;
+        return template.query(sql, reservationRowMapper(), date.toString());
+    }
+
+    @Override
+    public List<Reservation> findAll() {
+        String sql = """
+                SELECT r.id AS reservation_id, r.name, r.date, t.id AS time_id, t.start_at AS time_value
+                FROM reservation as r 
+                INNER JOIN reservation_time AS t
+                ON r.time_id = t.id
+                """;
+        return template.query(sql, reservationRowMapper());
+    }
+
+    @Override
+    public Reservation save(final Reservation reservation) {
+        String sql = "insert into reservation (name, date, time_id) values (?, ?, ?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        template.update(connection -> {
+            PreparedStatement ps = connection.prepareStatement(sql, new String[]{"id"});
+            ps.setString(1, reservation.getName());
+            ps.setString(2, reservation.getDate().toString());
+            ps.setLong(3, reservation.getTime().getId());
+            return ps;
+        }, keyHolder);
+
+        long id = keyHolder.getKey().longValue();
+        return reservation.toEntity(id);
+    }
+
+    @Override
+    public int deleteById(final long id) {
+        String sql = "delete from reservation where id = ?";
+        return template.update(sql, id);
+    }
+
+    @Override
+    public boolean existsByDateAndTime(final LocalDate date, final ReservationTime time) {
+        String sql = """
+                SELECT r.id AS reservation_id, r.name, r.date, t.id AS time_id, t.start_at AS time_value
+                FROM reservation as r 
+                INNER JOIN reservation_time AS t
+                ON r.time_id = t.id
+                WHERE r.date = ? and t.start_at = ?
+                """;
+        List<Reservation> result = template.query(sql, reservationRowMapper(), date.toString(),
+                time.getStartAt().toString());
+        return !result.isEmpty();
+    }
+
+    private RowMapper<Reservation> reservationRowMapper() {
+        return (rs, rowNum) -> {
+            ReservationTime reservationTime = ReservationTime.parse(rs.getString("time_value"))
+                    .toEntity(rs.getLong("time_id"));
```

Comment:
toEntity(id)를 받는 방식은 id가 없는 Entity에 저장 이후, Id를 할당해주기위한 방법으로 활용하기위함이었는데요. 

조회의 경우에는 Entity 객체 생성 시점에 id도 조회해올 수 있으니 생성자를 통해서 바로 주입해주는게 더 좋을 것 같아요. 😃

```suggestion
            ReservationTime reservationTime = new ReservationTime(
                rs.getLong("time_id"),
                rs.getString("time_value")
            );
```

Reply:
기존 메서드를 재활용할 생각만 하고, 새로 생성자를 통해 주입하는 방법을 생각하지 못했네요!
바꿔보겠습니다 감사합니다! 😄